manifests/install.pp

```diff
@@ -30,11 +30,26 @@
     extract_path => $aerospike::download_dir,
     creates      => $dest,
     cleanup      => $aerospike::remove_archive,
-  } ~>
-  exec { 'aerospike-install-server':
-    command     => "${dest}/asinstall",
-    cwd         => $dest,
-    refreshonly => true,
+    notify       => Exec['aerospike-install-server'],
```

Comment:
The problem you will have with a notify here is that if you have to change any of the parameters like the username or password here, even if the file doesn't change, you will have a reinstall happening potentially.